airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSourceOperations.java

```diff
@@ -4,17 +4,189 @@
+      case TINYTEXT, TEXT, MEDIUMTEXT, LONGTEXT, JSON, ENUM, SET -> putString(json, columnName, resultSet, colIndex);
+      case GEOMETRY -> json.put(columnName, Jsons.serialize(resultSet.getObject(colIndex)));
+      case NULL -> json.set(columnName, NullNode.instance);
+      default -> putDefault(json, columnName, resultSet, colIndex);
+    }
+  }
+
+  /**
+   * MySQL boolean is equivalent to tinyint(1).
+   */
+  @Override
+  protected void putBoolean(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index) throws SQLException {
     node.put(columnName, resultSet.getInt(index) == 1);
   }
 
+  @Override
+  public void setStatementField(final PreparedStatement preparedStatement,
+                                final int parameterIndex,
+                                final MysqlType cursorFieldType,
+                                final String value)
+      throws SQLException {
+    switch (cursorFieldType) {
+      case BIT -> setBit(preparedStatement, parameterIndex, value);
+      case BOOLEAN -> setBoolean(preparedStatement, parameterIndex, value);
+      case TINYINT, TINYINT_UNSIGNED, SMALLINT, SMALLINT_UNSIGNED, MEDIUMINT, MEDIUMINT_UNSIGNED -> setInteger(preparedStatement, parameterIndex,
+          value);
+      case INT, INT_UNSIGNED, BIGINT, BIGINT_UNSIGNED -> setBigInteger(preparedStatement, parameterIndex, value);
+      case FLOAT, FLOAT_UNSIGNED, DOUBLE, DOUBLE_UNSIGNED -> setDouble(preparedStatement, parameterIndex, value);
+      case DECIMAL, DECIMAL_UNSIGNED -> setDecimal(preparedStatement, parameterIndex, value);
+      case DATE -> setDate(preparedStatement, parameterIndex, value);
+      case DATETIME, TIMESTAMP -> setTimestamp(preparedStatement, parameterIndex, value);
+      case TIME -> setTime(preparedStatement, parameterIndex, value);
+      case YEAR, CHAR, VARCHAR, TINYTEXT, TEXT, MEDIUMTEXT, LONGTEXT, ENUM, SET -> setString(preparedStatement, parameterIndex, value);
+      case TINYBLOB, BLOB, MEDIUMBLOB, LONGBLOB, BINARY, VARBINARY -> setBinary(preparedStatement, parameterIndex, value);
+      // since cursor are expected to be comparable, handle cursor typing strictly and error on
+      // unrecognized types
+      default -> throw new IllegalArgumentException(String.format("%s is not supported.", cursorFieldType));
+    }
+  }
+
+  /**
+   * Convert ambiguous MySQL types to more specific ones, so that later on we can correctly map them
+   * to {@link JsonSchemaPrimitive}. TODO (liren): Currently, for columns like CHAR, the field json
+   * does not contain information about its charset, which is needed to determine whether the column
+   * is a string or binary. We don't distinguish between string vs binary in
+   * {@link JsonSchemaPrimitive} for now. So it is fine. However, in the future, if we want to
+   * separate these two types, we need to update the column metadata query for MySQL. See
```

Comment:
should we have an issue for this?

Reply:
Added an issue: https://github.com/airbytehq/airbyte/issues/8723

Reply:
Actually JDBC does return `BINARY` for `CHAR` columns with `binary` character set. Same for `VARCHAR`. So this is a false alarm.